### PR TITLE
keystore: unit test and harden unlock()

### PR DESCRIPTION
**Pull request by benma**
Be able to use it as a PW check only.

### CODE REVIEW

File: src/keystore.c [hunk 1]
```diff
@@ -292,8 +297,15 @@ keystore_error_t keystore_unlock(const char* password, uint8_t* remaining_attemp
         return KEYSTORE_ERR_GENERIC;
     }
     if (password_correct) {
-        memcpy(_retained_seed, seed, sizeof(_retained_seed));
-        _is_unlocked_device = true;
+        if (_is_unlocked_device) {
+            // Already unlocked. Fail if the seed changed under our feet (should never happen).
```
Comment (NickeZ) [inline src/keystore.c]: should we have an abort if it should never happen?

File: src/workflow/restore.c [hunk 2]
```diff
@@ -40,13 +40,11 @@ static bool _result = false;
 static bool _restore(const char* password)
 {
     bool res = restore_seed(&_backup_data, password);
-    component_t* status;
     if (res) {
```
Comment (NickeZ) [inline src/workflow/restore.c]: Maybe `res == true`?

File: src/keystore.c (hunk 1 above)
Comment (benma) [reply src/keystore.c]: Maybe. In this case, it was a bit easier to unit test, and the caller Aborts() anyway on a generic error.

File: src/workflow/restore.c (hunk 2 above)
Comment (benma) [reply src/workflow/restore.c]: hmm why?

**Pull request merged**